docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+## Do include appropriate `package.json` entries
+
+Your `package.json` should both its `main` entrypoint and `module` entrypoint to the same es module file. You will find this already configured for you by `npm init @open-wc`, but ensure your `package.json` file includes something simialar to the following:
+
+```
+{
+  "main": "index.js",
+  "module": "index.js",
+  // ...
+}
+```
+
+## Do export element classes
+
+Every custom element is a class extension of `HTMLElement`; e.g. `class MyElement extends HTMLElement {}`. Be sure that you export this class from your package. When starting from `npm init @open-wc` you will see this export in `src/MyElement.js` and reported by the `index.js` entry point. This will make it possible for both customization of your element via extension (e.g. `class NewElement extends MyElement {}`) as well as for advanced usage of the custom element in Scoped Registries, both [native](https://github.com/w3c/webcomponents/issues/716) (when available) or [synthetic](/scoped-elements/).
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when script effects state outside of the scope in which it is run. This includes when code in an es module effects values available on the `window` or `document` as is inherent to the reality of Custom Elements in that `window.customElements.define()` registers an element in the global registry attached to `window`, whether you explicitly include `window` in your code when doing so or not. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` exemplifies this through its creation of `my-element.js` which manages the registration of the custom element that this command generates.
+
+## Do keep side effects out of your index.js
+
+When not able to keep code with side effects out of your entry point, be sure to contain such commands within a method (e.g. `run()`, `init()`, etc.) to prevent those side effects from occurring at run time.
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure your package as you do. To prevent dependency pointing from causing issue across different environments be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers, always
+
+When not pointing directly the entry point files referenced by "bar" import specifiers, be sure to reference files (both in your package and your dependencies) with their complete file extension; e.g. `import { ifDefined } from 'lit-html/directives/if-defined.js';`
```

Comment:
> When not pointing directly the entry point files referenced by "bar" import specifiers,

this sentence is unclear to me

Reply:
Trying to say something in reference when not requesting the entry point file specifically (via "bare" import specifiers)...would love thoughts on dumbing it down, because it's kind of a confusing concept in general.

Reply:
When importing something that doesn't point directly to the _entry point_ of a module, e.g.: `import {LitElement} from 'lit-element';`, *always* make sure to include a file extension. `import { ifDefined } from 'lit-html/directives/if-defined.js';`

Maybe another do/dont list?

✅ Do:
`import { html } from 'lit-html';`
`import { ifDefined } from 'lit-html/directives/if-defined.js';`

❌ Dont:
`import { html } from 'lit-html.js';`
`import { ifDefined } from 'lit-html/directives/if-defined';`